### PR TITLE
Use instance UID instead of name in broker op backoff/retry

### PR DESCRIPTION
**Pull request by jboyd01**
fixes https://github.com/kubernetes-incubator/service-catalog/issues/2279

Use K8s metadata UID instead of instance name
Delete the instance entry from the retryMap on successful deprovisioning
Clarify debug logging

Either fix 1 or 2 would work, the combination looks like the correct approach.

### CODE REVIEW

File: pkg/controller/controller_instance.go [hunk 1]
```diff
@@ -115,8 +115,8 @@ type backoffEntry struct {
 type instanceOperationBackoff struct {
 	// lock to be used for accessing retry map
 	mutex       sync.RWMutex
-	instances   map[string]backoffEntry
-	rateLimiter workqueue.RateLimiter // used to calculate next retry time
+	instances   map[string]backoffEntry // Key is instance UUID
+	rateLimiter workqueue.RateLimiter   // used to calculate next retry time, Key is instance UUID
```
Comment (MHBauer) [inline pkg/controller/controller_instance.go]: I would say 'kubernetes metadata UID' for these, right?

Comment (jboyd01) [reply pkg/controller/controller_instance.go]: yes, I updated the comment.

**Comment by nilebox** [comment]
@jboyd01 thanks for fixing this so quickly!

**Pull request merged**